torch/fx/experimental/proxy_tensor.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import functools
+from typing import Any, Dict, Optional, Tuple, Callable, Union
+import torch
+from torch._C import _disabled_torch_function_impl
+import torch.utils._pytree as pytree
+from torch.fx import Tracer, GraphModule
+import torch.fx as fx
+from torch.fx.passes.shape_prop import _extract_tensor_metadata
+from contextlib import contextmanager
+
+aten = torch.ops.aten
+
+CURRENT_DECOMPOSITION_TABLE = {}
+
+
+@contextmanager
+def no_dispatch():
```

Comment:
Use the one from `torch.testing._internal.logging_tensor.no_dispatch` ?

Reply:
that's a terrible location for it.

The root of our evils is we don't have a "torch dispatch" module. I propose torch.dispatch. Any objections?

Reply:
Well this function should not exist?
Once the mode are fixed by Sam and users can call super(). There is no need for this function at all!

I do agree that we can have this namespace if needed though if we don't want to put it in `torch.overrides` (which has an ok name to house both torch_function and torch_dispatch).

Reply:
Well, we need this function to exist in some form for internal implementation purposes (e.g., actually implementing super())

I'm not very hot on torch.overrides because although the name is generic it really is very torch_function leaning right now.

Reply:
> I'm not very hot on torch.overrides because although the name is generic it really is very torch_function leaning right now.

Sure, but we could change that? Also I plead guilding of adding `torch.overrides.enable_reentrant_dispatch()` there. But we can move it somewhere else if we prefer a new namespace.